GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/SendReminderEmailsJob.cs

```diff
@@ -170,18 +201,50 @@ private void CheckUserAndSendReminderEmailsForSectorTypeAndReportingYear(User us
                 sectorType: sectorType.ToString().ToLower());
         }
 
-        private static void SaveReminderEmailRecord(User user, SectorTypes sectorType, bool wasAnEmailSent)
+        private ReminderEmail GetReminderEmailRecord(User user, SectorTypes sectorType, DateTime reminderDate)
         {
-            var reminderEmailRecord = new ReminderEmail
+            var reminderEmailRecord = dataRepository
+                .GetAll<ReminderEmail>()
+                .FirstOrDefault(
+                    re => re.UserId == user.UserId
+                          && re.ReminderDate.HasValue
+                          && re.ReminderDate.Value.Date == reminderDate.Date
+                          && re.SectorType == sectorType);
+
+            if (reminderEmailRecord == null)
             {
-                UserId = user.UserId,
-                SectorType = sectorType,
-                DateChecked = VirtualDateTime.Now,
-                EmailSent = wasAnEmailSent
-            };
-
-            var dataRepository = Global.ContainerIoC.Resolve<IDataRepository>();
-            dataRepository.Insert(reminderEmailRecord);
+                reminderEmailRecord = new ReminderEmail
+                {
+                    UserId = user.UserId,
+                    SectorType = sectorType,
+                    DateChecked = VirtualDateTime.Now,
+                    ReminderDate = reminderDate,
+                    EmailSent = false,
+                    Status = ReminderEmailStatus.InProgress
+                };
+                dataRepository.Insert(reminderEmailRecord);
+            }
+            else if (reminderEmailRecord.Status == ReminderEmailStatus.InProgress
+                     && reminderEmailRecord.DateChecked < VirtualDateTime.Now.AddMinutes(-15))
+            {
+                reminderEmailRecord.DateChecked = VirtualDateTime.Now;
+            }
+            else
+            {
+                return null;
```

Comment:
Maybe we could separate this function into two that are called by the parent function one after the other?  The first one could be "getReminderEmailRecordOrCreateIfNull". This function would call the "getReminderEmailRecord" function and then, if null, call the "createReminderEmailRecord" function. After the "getReminderEmailRecordOrCreateIfNull" function, we would have an if statement that would be something like:
```
if (reminderEmailSendingProcessHasNotStartedOrHasExpired)" {
   updateReminderEmailCheckedDate()
   CheckUSerAndSendReminderEmailsForSectorTypeAndReportingYear
}
```

Reply:
The new created reminder emails will never pass this condition (only after they expire)